### ISSUE TITLE
Bug? After starting the PaymentActivity, Android may decide to  close the parent activity

### ISSUE THREAD

**Issue by mmattini** [Adyen/adyen-checkout-android#22]
Hi 
Below is an extract from the sample app. In MainActivity:
 **Intent intent = new PaymentActivity.PaymentActivityBuilder(checkoutRequest).build(this, context);
 startActivity(intent);**

In the example, the MainActivity implements the AdyenCheckoutListener interface, and once payment is finished, code in the MainActivity will be executed (at least this is what the sample says).

The point is that once you started a child activity (e.g. PaymentActivity), Android might decide to close your parent activity (e.g MainActivity) at any time, if system requires more memory. So there is  no guarantee that the listener code in MainActivity will run. In fact the app may crash.

In summary I am not really comfortable with this pattern. Is there a better way to start the PaymentActivity?

Thanks
Max
